app/database/admin/data_operator/operators.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Q} from '@nozbe/watermelondb';
+import Model from '@nozbe/watermelondb/Model';
+
+import App from '@typings/database/app';
+import CustomEmoji from '@typings/database/custom_emoji';
+import {
+    DataFactory,
+    RawApp,
+    RawCustomEmoji,
+    RawGlobal,
+    RawRole,
+    RawServers,
+    RawSystem,
+    RawTermsOfService,
+} from '@typings/database/database';
+import Global from '@typings/database/global';
+import Role from '@typings/database/role';
+import Servers from '@typings/database/servers';
+import System from '@typings/database/system';
+import TermsOfService from '@typings/database/terms_of_service';
+
+import {OperationType} from './index';
+
+/**
+ * operateAppRecord: Prepares record of entity 'App' from the DEFAULT database for update or create actions.
+ * @param {Database} db
+ * @param {OperationType} optType
+ * @param {string} tableName
+ * @param {RecordValue} value
+ * @returns {Promise<void>}
+ */
+export const operateAppRecord = async ({db, optType, tableName, value}: DataFactory) => {
```

Comment:
Should tableName be omitted as parameter and correct value just passed directly to `operateBaseRecord` since the name of function suggests it should be a particular table?

Reply:
If table name needs to be passed, please add tests that exercise incorrect table names passed for all the operators

Reply:
I'm now enforcing the correct table name via the use of Typescript.

Reply:
Hi @avinashlng1080 Can you please add tests that checks if correct table name is enforced for all operators?